File: Encapsulation/Java/Personal_Information/src/Identity/Person.java
package Identity;

public class Person extends Information{
    private String name; //Variable used to keep to keep track of the person's name

    public Person(){ //Default constructor
        super();
        name = " ";
    }
    public Person(String social_security, String address, String name){ //Overloaded constructor
        super(social_security,address);
        this.name = name;
    }

    public void SetName(String name){ //Name setter
        this.name = name;
    }
    public String GetName(){ //Name getter
        return this.name;
    }
}

File: Encapsulation/C++/Personal_Information/Person.cpp
#include<iostream>
#include "Personal_Information.cpp"
using namespace std;

//Objective create a sub class from Personal_Information to show access modifiers
class Person : public Personal_Information{
private:
    string name; //Variable used to keep track of the name

public:
    Person():Personal_Information(){ //Default constructor
        name = "";
    }
    //Overloaded constructor
    Person(string social_security, string address, string name):Personal_Information(social_security, address){
        this->name = name;
    }
    void SetName(string name){this->name = name;} //Name setter
    string GetName(){return this->name;} //Name getter
};